browser_use/agent/service.py

```diff
@@ -396,27 +394,78 @@ def _setup_action_models(self) -> None:
 		self.DoneActionModel = self.controller.registry.create_action_model(include_actions=['done'])
 		self.DoneAgentOutput = AgentOutput.type_with_custom_actions(self.DoneActionModel)
 
-	def _set_tool_calling_method(self) -> ToolCallingMethod | None:
-		tool_calling_method = self.settings.tool_calling_method
-		if tool_calling_method == 'auto':
-			if is_model_without_tool_support(self.model_name):
-				return 'raw'
-			elif self.chat_model_library == 'ChatGoogleGenerativeAI':
-				return None
-			elif self.chat_model_library == 'ChatOpenAI':
-				return 'function_calling'
-			elif self.chat_model_library == 'AzureChatOpenAI':
-				# Azure OpenAI API requires 'tools' parameter for GPT-4
-				# The error 'content must be either a string or an array' occurs when
-				# the API expects a tools array but gets something else
-				if 'gpt-4' in self.model_name.lower():
-					return 'tools'
-				else:
-					return 'function_calling'
+	def _test_tool_calling_method(self, method: str) -> bool:
+		"""Test if a specific tool calling method works with the current LLM."""
+		try:
+			# Create a simple test message
+			test_message = HumanMessage(content='Hello, world!')
+
+			if method == 'raw':
+				# For raw mode, we just check if we can invoke the model
+				response = self.llm.invoke([test_message])
+				# Basic validation of response
+				if not response or not hasattr(response, 'content'):
+					return False
+				return True
 			else:
-				return None
-		else:
-			return tool_calling_method
+				# For other methods, try to use structured output
+				structured_llm = self.llm.with_structured_output(self.ActionModel, include_raw=True, method=method)
+				# Try a simple invocation to test the method
+				response = structured_llm.invoke([test_message])
+				# TODO: add validation of response
```

Comment:
can you try and keep the capital of France question + answer check here as part of the tool call prompt & validation logic? we've found it very useful in practice to set a minimum intelligence bar for models, it really reduces the support load from people trying to run very low-power or broken models